### ISSUE TITLE
Support for attaching multiple cdrom devices to VMs

### ISSUE THREAD

**Issue by mrmagooey** [ansible/ansible#45407]
##### SUMMARY
Currently only a single cdrom can be attached to a vm using the vmware_guest.py module. There are instances where it would be useful to attach multiple cdrom devices, e.g. Some operating systems allow for automated install files to be inserted as cdroms into the machine during initial install. Without the ability to have multiple cdroms attach to a vm on startup, the ability to make use of this auto-install ability is unavailable (as the install iso takes up the only available cdrom  device).

Similar issue #44820 

##### ISSUE TYPE
Allow multiple cdroms to be attached to a vm.

##### COMPONENT NAME
cloud/vmware/vmware_guest.py

##### ADDITIONAL INFORMATION
The only difference would be that the cdrom section would become a list, going from what it is currently:

      cdrom:
        type: iso
        iso_path: "[{{ vmware_isostore }}] Windows Server 2012 R2.ISO"

To being

      cdrom:
      - type: iso
         iso_path: "[{{ vmware_isostore }}] Windows Server 2012 R2.ISO"
      cdrom:
      - type: iso
         iso_path: "[{{ vmware_isostore }}] my.iso"

**Comment by ansibot** [comment]
Files identified in the description:
* [lib/ansible/modules/cloud/vmware/vmware_guest.py](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/vmware/vmware_guest.py)
* [lib/ansible/playbook/become.py](https://github.com/ansible/ansible/blob/devel/lib/ansible/playbook/become.py)

If these files are inaccurate, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@mrmagooey Greetings! Thanks for taking the time to open this issue. In order for the community to handle your issue effectively, we need a bit more information. 

Here are the items we could not find in your description:
- ansible version

Please set the description of this issue with this template:
https://raw.githubusercontent.com/ansible/ansible/devel/.github/ISSUE_TEMPLATE.md

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
cc @Akasurde @dav1x @nerzhul @pdellaert @warthog9
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
Hi @mrmagooey,

Thank you for the issue, just so you are aware we have a dedicated Working Group for vmware.
You can find other people interested in this in #ansible-vmware on Freenode IRC
For more information about communities, meetings and agendas see https://github.com/ansible/community

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by mrmagooey** [comment]
!component -lib/ansible/playbook/become.py

**Comment by ansibot** [comment]
Files identified in the description:
* [lib/ansible/modules/cloud/vmware/vmware_guest.py](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/vmware/vmware_guest.py)

If these files are inaccurate, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
cc @ckotte
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by Tomorrow9** [comment]
There is pull request to address this issue: https://github.com/ansible/ansible/pull/37869, but it's closed.

**Comment by Tomorrow9** [comment]
I'll take care of this in a new PR later, thanks.

**Comment by Tomorrow9** [comment]
@Akasurde do you think it's a good plan that I implement this as the way in #49534, since there are also "IDE controller" and "SATA controller" need to be thought about for CD/DVD drive? And I'll add support for   "IDE controller" this time, after #49534 merged I'll add support for "SATA controller", since disk may use "SATA controller" too.
cdrom:
    - controller_type: ide
      controller_number: 0
      unit_number: 0
      type: client
    - controller_type: sata
      controller_number: 1
      unit_number: 0
      type: iso
      iso_path: "[test_path] test.iso"

**Comment by Tomorrow9** [comment]
@mrmagooey, Hi, I submitted the changes, can you try it and if there is any issue please let me know. Thanks. It supports IDE controller now and will add SATA controller later.

**Comment by ansibot** [comment]
cc @lparkes
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by spongebob331** [comment]
Any idea when this is coming?  I could use the multiple cdrom feature, as well.

**Comment by ansibot** [comment]
cc @goneri
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
cc @pgbidkar
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)
